### ISSUE TITLE
Can't transfer colon from AppleScript

### ISSUE THREAD

**Issue by AttracksPilot123** [joncampbell123/dosbox-x#4549]
### Question

Hi there,

With my previous iMac I used to run an AppleScript to launch and manage Dosbox.
Since I made the transition to MacBook Air (2023) with Apple Silicon, that doesn't run anymore... of course.
So I downloaded Dosbox-X. Thanks for that! And adjusted the script a little.

But I can't transmit a simple colon (:), which is absolutely necessary to change diectory.
It appears as a semicolon (;).
I've tried a lot of different key-combinations... but still no success.
Can you help me, please?

Cheers!

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by AttracksPilot123** [comment]
It would be very helpful if some of you is willing to react on my question.

**Comment by AttracksPilot123** [comment]
polite bump...

**Comment by AttracksPilot123** [comment]
Any news?

**Comment by joncampbell123** [comment]
I don't know... is Apple translating things in the way? I don't think anything changed there with Mac OS.

**Comment by AttracksPilot123** [comment]
Thanks for your comment.
Perhaps it is an issue at the side of Apple. I'll check it out.
Cheers!

**Comment by AttracksPilot123** [comment]
I tried to test it, but it seems to occur during the transfer from AppleScript to DropBox-x.
![Uploading Screenshot 2024-02-16 at 09.45.28.png…]()

**Comment by AttracksPilot123** [comment]
![Screenshot 2024-02-16 at 09 45 28](https://github.com/joncampbell123/dosbox-x/assets/148647124/a01ae2bd-1521-4ea1-b06c-63cbd6854b99)

**Comment by maron2000** [comment]
I think you need to let DOSBox-X that the Shift-key is pressed.
Colon is `Shift + ;` if you're using US keyboards.

**Comment by AttracksPilot123** [comment]
Thanks for your help.
As you can see at the left side, I entered a ":" (not a ";").
Or do you mean something else?

**Comment by maron2000** [comment]
I know that you entered a ':' but the problem is that what keystroke is actually given to DOSBox-X.
We already know that entering ':' doesn't work, try explicitly entering `Shift+;`, there is a AppleScript command to press modifier keys.
BtW, are you using a US keyboard?

**Comment by maron2000** [comment]
I think my assumption is correct since in you screenshot, the path of the mount command is `/Users/...` but the letters input in DOSBox-X are all lower case, which means Shift key isn't sent (or DOSBox-X fails to receive it, I don't know).

**Comment by AttracksPilot123** [comment]
I'm using a US keyboard indeed. And I tried to insert both key codes for Shift (57 en 60), but with no result.

**Comment by maron2000** [comment]
I think this is a problem of SDL library. Maybe a difference in SDL1 & 2 if you haven't tried.
You may also want to try pasting text via clipboard.
https://github.com/joncampbell123/dosbox-x/wiki#dosbox-xs-special-keys
https://github.com/joncampbell123/dosbox-x/wiki/Guide%3AClipboard-support-in-DOSBox%E2%80%90X

**Comment by AttracksPilot123** [comment]
Yes, if I enter the “:” directly in DOSBox-X, it’s perfectly alright.
I guess AppleScript is the culprit, because if I record my actions and play it afterwards, everything is copied fine.

Thanks for your help!
Gert Jan

**Comment by AttracksPilot123** [comment]
With Automator I mean, but that is not failsafe.
